src/components/split-axis-events.js

```diff
@@ -1,30 +1,376 @@
-AFRAME.registerComponent("split-axis-events", {
+const angleTo4Direction = function(angle) {
+  angle = (angle * THREE.Math.RAD2DEG + 180 + 45) % 360;
+  if (angle > 0 && angle < 90) {
+    return "north";
+  } else if (angle >= 90 && angle < 180) {
+    return "west";
+  } else if (angle >= 180 && angle < 270) {
+    return "south";
+  } else {
+    return "east";
+  }
+};
+
+const angleTo8Direction = function(angle) {
+  angle = (angle * THREE.Math.RAD2DEG + 180 + 45) % 360;
+  var direction = "";
+  if ((angle >= 0 && angle < 120) || angle >= 330) {
+    direction += "north";
+  }
+  if (angle >= 150 && angle < 300) {
+    direction += "south";
+  }
+  if (angle >= 60 && angle < 210) {
+    direction += "west";
+  }
+  if ((angle >= 240 && angle < 360) || angle < 30) {
+    direction += "east";
+  }
+  return direction;
+};
+
+AFRAME.registerComponent("dpad-as-axes", {
+  schema: {
+    inputName: { default: "dpad" },
+    name: { default: "dpad_axes" },
+    emitter: { default: "#left-hand" }
+  },
+
+  init: function() {
+    this.mapping = [
+      {
+        direction: "north",
+        axes: [0, 1]
+      },
+      {
+        direction: "northeast",
+        axes: [1, 1]
+      },
+      {
+        direction: "east",
+        axes: [1, 0]
+      },
+      {
+        direction: "southeast",
+        axes: [1, -1]
+      },
+      {
+        direction: "south",
+        axes: [0, -1]
+      },
+      {
+        direction: "southwest",
+        axes: [-1, -1]
+      },
+      {
+        direction: "west",
+        axes: [-1, 0]
+      },
+      {
+        direction: "northwest",
+        axes: [-1, 1]
+      }
+    ];
+    this.handlers = [];
+  },
+
+  play: function() {
+    var inputName = this.data.inputName;
+    for (var pair of this.mapping) {
+      this.handlers[pair.direction] = this.emitAxes(pair.axes).bind(this);
+      this.el.addEventListener(
+        `${inputName}_${pair.direction}`,
+        this.handlers[pair.direction]
+      );
+    }
+  },
+
+  pause: function() {
+    var inputName = this.data.inputName;
+    for (var pair of this.mapping) {
+      this.el.removeEventListener(
+        `${inputName}_${pair.direction}`,
+        this.handlers[pair.direction]
+      );
+    }
+  },
+
+  emitAxes: function(axes) {
+    const name = this.data.name;
+    const inputName = this.data.inputName;
+    const emitter = document.querySelector(this.data.emitter);
+    return function(event) {
+      emitter.emit(name, { axis: [axes[0], axes[1]] });
+    };
+  }
+});
+
+AFRAME.registerComponent("wasd-dpad", {
```

Comment:
Generally, I think we should have a limit of one component per source file, IMO.